app/services/strategy_testing_service.py

```
@@ -0,0 +1,629 @@
+    async def _get_daily_ohlcv(self, symbol: str, date: datetime) -> Optional[Dict[str, Any]]:
+        """Get daily OHLCV data for specific date."""
+        try:
+            # This would call historical data APIs
+            # For now, generate realistic historical data
+            import random
+            
+            # Base price with realistic movement
+            base_price = 50000 if symbol == "BTC" else 3000 if symbol == "ETH" else 100
+            price_variation = base_price * 0.05  # 5% daily variation
+            
+            open_price = base_price + (random.random() - 0.5) * price_variation
```

Ruff (0.12.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)